core/transformer/src/IModelImporter.ts

```diff
@@ -471,3 +492,15 @@ function hasNavigationValueChanged(navigationProperty1: any, navigationProperty2
 function hasValueChanged(property1: any, property2: any): boolean {
   return JSON.stringify(property1) !== JSON.stringify(property2);
 }
+
+/** check if element props are a subcategory */
+function isSubCategory(props: ElementProps): props is SubCategoryProps {
+  return props.classFullName === SubCategory.classFullName;
+}
+
+/** check if element props are a subcategory without loading the element */
+function isDefaultSubCategory(props: SubCategoryProps): boolean {
+  if (props.parent?.id === undefined)
+    throw new IModelError(IModelStatus.BadElement, "subcategory had no parent");
```

Comment:
This error message has very little information.  It should be enhanced so someone could understand what subcategory had no parent without debugging into the code

Reply:
updates, please take a look